lib/cc/pull_requests.rb

```diff
@@ -1,8 +1,28 @@
 class CC::PullRequests < CC::Service
+  ABLE_TO_UPDATE_STATUS_MESSAGE = "Access token is valid"
+  ABLE_TO_UPDATE_STATUS_AND_COMMENT_MESSAGE = "Access token is valid"
+  ABLE_TO_UPDATE_STATUS_BUT_NOT_COMMENT_MESSAGE = "OAuth token requires 'repo' scope to post comments."
   def receive_test
     setup_http
 
-    receive_test_status
+    # this will raise an HTTPError or be successful:
+    able_to_update_status_response = check_if_able_to_update_status
+
+    if welcome_comment_implemented? && config.welcome_comment_enabled
```

Comment:
The structure here feels confusing in that we're treating the status test & the comment test quite differently. Can we make these more similar? Is it problematic that we're not raising an exception for comment failure the same as status failure? This might be a good problem to pair on.